### ISSUE TITLE
数据集问题

### ISSUE THREAD

**Issue by ydcs1** [liyongqi67/ALPINE#2]
你好，有两个问题麻烦作者解答一下：
1.下载的数据集中items的数量和论文上的不一样
2.放出的数据集没有测试集只有训练集

**Comment by liyongqi67** [comment]
1、visual64_select中含有3242315个items的视觉特征，但其中只有3239534个是和user产生了交互的
2、数据集放出了train和test集，但当时为了和之前的一个很相关的工作，也就是论文中dataset II保持一致，构建这个数据集的时候没有构建validation set（现在想来这样的设定并不好）。我在代码中写的val_interaction_data其实就是test集，我把这个地方在代码中修正一下。

**Comment by ydcs1** [comment]
感谢学长回复
但我有点没明白：
1.我怎么筛选出这些3239534个交互过的视觉特征，这些特征里有啥标志嘛
2.还有这个val_interaction_data和train_interaction_data加载的文件是一样的欸，而且"dataset.pkl"里面只有10,931,092，和论文给出
的训练集数量是吻合的，所以还有2,730,291的测试集数据没找到。


望学长解答！







在 2020-06-26 15:33:00，"liyongqi" <notifications@github.com> 写道：

1、visual64_select中含有3242315个items的视觉特征，但其中只有3239534个是和user产生了交互的
2、数据集放出了train和test集，但当时为了和之前的一个很相关的工作，也就是论文中dataset II保持一致，构建这个数据集的时候没有构建validation set（现在想来这样的设定并不好）。我在代码中写的val_interaction_data其实就是test集，我把这个地方在代码中修正一下。

**Comment by ydcs1** [comment]
1.前面一封邮件第一个问题应该是根据"dataset.pkl"这个数据集筛选出交互过的数据吧
2.另一个问题是图中红色的两列代表啥意思

谢谢！










在 2020-06-26 15:33:00，"liyongqi" <notifications@github.com> 写道：

1、visual64_select中含有3242315个items的视觉特征，但其中只有3239534个是和user产生了交互的
2、数据集放出了train和test集，但当时为了和之前的一个很相关的工作，也就是论文中dataset II保持一致，构建这个数据集的时候没有构建validation set（现在想来这样的设定并不好）。我在代码中写的val_interaction_data其实就是test集，我把这个地方在代码中修正一下。

**Comment by liyongqi67** [comment]
1、dataset.pkl里的train_interaction_data是user和item interaction的训练数据，test_interaction_data是user和item interaction的测试数据，len(train_interaction_data)=10931092, len(test_interaction_data)=2730291
2、train_interaction_data里面涉及到的video_id有2734430个，test_interaction_data里包含的video_id有871070，总共video_id数目为3239534.(说明test_interaction_data里的video有的在训练集出现过）
3、train_interaction_data和test_interaction_data里有video_id，可以通过train_interaction_data[i][0][1]和test_interaction_data[i][0][1]进行读取，然后这些id就是整个训练测试所有的video_id，然后可以通过这些id去过滤visual64_select，如果你想的话
4、下面是统计数据的代码

import heapq
import random
import math
import tensorflow as tf
import numpy as np
import os
import io
import time
import datetime
from tensorflow.contrib import learn
from tqdm import tqdm
from sklearn.metrics import roc_auc_score
from sklearn.decomposition import PCA 
import sys

import pickle



visual="/home/chengyao/data_dir/visual64_select.npy"
visual=np.load(visual)
print(visual.shape)

with open("/home/chengyao/data_dir/dataset.pkl", 'rb') as f:
  train_interaction_data= pickle.load(f)
  test_interaction_data= pickle.load(f)
  pos_his_data= pickle.load(f)
  neg_his_data=pickle.load(f)
  pos_edge_data=pickle.load(f)
  neg_edge_data=pickle.load(f)


item_id={}
for s in train_interaction_data:
  if(s[0][1] not in item_id):
    item_id[s[0][1]]=1
for s in test_interaction_data:
  if(s[0][1] not in item_id):
    item_id[s[0][1]]=1


print("# items",len(item_id))

print("#interactions in training set",len(train_interaction_data))
print("#interactions in test set",len(test_interaction_data))

**Comment by ydcs1** [comment]
你好，我输出了"dataset.pkl"的结果，dataset是list类型，并没有train_interaction_data和test_interaction_data










在 2020-06-26 19:40:05，"liyongqi" <notifications@github.com> 写道：

1、dataset.pkl里的train_interaction_data是user和item interaction的训练数据，test_interaction_data是user和item interaction的测试数据
2、train_interaction_data里面涉及到的video_id有2734430个，test_interaction_data里包含的video_id有871070，总共video_id数目为3239534.(说明test_interaction_data里的video有的在训练集出现过）
3、train_interaction_data和test_interaction_data里有video_id，可以通过train_interaction_data[i][0][1]进行读取

**Comment by liyongqi67** [comment]
这样读取，这在源代码里有
with open("/home/chengyao/data_dir/dataset.pkl", 'rb') as f:
train_interaction_data= pickle.load(f)
test_interaction_data= pickle.load(f)
pos_his_data= pickle.load(f)
neg_his_data=pickle.load(f)
pos_edge_data=pickle.load(f)
neg_edge_data=pickle.load(f)

**Comment by ydcs1** [comment]
![捕获](https://user-images.githubusercontent.com/62939900/85861250-9cde3b80-b7f2-11ea-93c4-50d42ea7f476.PNG)

最后问一下最后两列是啥意思。

**Comment by liyongqi67** [comment]
序列的实际长度，feed到model做mask用的
